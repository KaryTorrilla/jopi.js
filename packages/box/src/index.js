import styled from 'styled-components'
import {
  space,
  color,
  layout,
  flexbox,
  border,
  typography,
  position,
  variant,
} from 'styled-system'
import css from '@styled-system/css'

const sx = props => css(props.sx)(props.theme)
const base = props => css(props.__css)(props.theme)
const theme = ({ tx = 'variants' }) =>
  variant({
    scale: tx,
  })

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    margin: 0,
    minWidth: 0,
  },
  base,
  theme,
  sx,
  space,
  color,
  layout,
  flexbox,
  typography,
  border,
  position
)

export const Flex = styled(Box)({
  display: 'flex',
})
