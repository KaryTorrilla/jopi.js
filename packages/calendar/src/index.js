import React from 'react'
import FullCalendar from '@fullcalendar/react' // must go before plugins
import dayGridPlugin from '@fullcalendar/daygrid' // a plugin!
import '@oneloop/theme/styles/globals.css'
import '../styles/styles.css'

export const Calendar = () => {
  /*   const calendarOptions = {
      events: [],
      plugins: [timeGridPlugin],
      initialView: 'timeGridDay',
      nowIndicator: true,
      headerToolbar: {
        start: 'prev',
        center: 'title',
        end: 'next',
      },
      editable: true,
      eventResizableFromStart: true,
      dragScroll: true,
      dayMaxEvents: 2,
      navLinks: true,
    } */

  return (
    <div className='calendarContainer'>
      <FullCalendar
        plugins={[dayGridPlugin]}
        initialView="dayGridMonth"
      />
    </div>
  )
}
