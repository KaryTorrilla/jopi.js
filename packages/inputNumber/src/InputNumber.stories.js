import React, { useState } from 'react'
import { InputNumber } from '.'
import { Text } from '@oneloop/text'

export default {
  component: InputNumber,
  title: 'InputNumber',
  argTypes: {
    variant: {
      name: 'variant',
      description: "**Variantes de color disponibles:** [ 'default']",
      type: 'text',
      control: { type: 'none' },
    },
    max: {
      name: 'max',
      description: 'Número máximo del input',
      type: 'number',
      control: { type: 'none' },
    },
    min: {
      name: 'min',
      description: 'Número mínimo del input',
      type: 'number',
      control: { type: 'none' },
    },
    startVal: {
      name: 'startVal',
      description: 'Valor inicial',
      type: 'number',
      control: { type: 'none' },
    },
    step: {
      name: 'step',
      description: 'Intervalo para agregar y restar',
      type: 'number',
      control: { type: 'none' },
    },
    val: {
      name: 'val',
      description:
        'Recibe la función donde se quiera guardar el valor mostrado',
      type: 'function',
      control: { type: 'none' },
    },
    text: {
      name: 'text',
      description: 'Texto de la etiqueta',
      type: 'string',
      control: { type: 'none' },
    },
    error: {
      name: 'error',
      description: 'Booleano para indicar que el campo falta de completar',
      type: 'boolean',
      control: { type: 'none' },
    },
  },
}

export const Minutes = () => {
  const [value, setValue] = useState(0)
  return (
    <>
      <InputNumber val={setValue} text="Minutes before" />
      <Text>{value}</Text>
    </>
  )
}
