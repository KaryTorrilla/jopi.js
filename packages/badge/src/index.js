import React from 'react'
import { Box } from '@oneloop/box'
import '@oneloop/fonts'

export const Badge = ({ isMedium, hasIcon, isNotButton, ...props }) => (
  <Box
    as={ isNotButton ? "" : "button" }
    tx="buttons"
    variant="primary"
    {...props}
    __css={{
      lineHeight: '14px',
      height: isMedium ? '22px' : '16px',
      display: 'inline-block',
      fontSize: '10px',
      fontFamily: 'primary',
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      letterSpacing: '0.05em',
      px: 2,
      py: '2px',
      color: 'white',
      bg: 'primary',
      border: 0,
      borderRadius: isMedium ? '14px' : '8px',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: hasIcon ? '4px 8px' : '1px 5px',
      position: 'static',
      flex: 'none',
      order: 0,
      flexGrow: 0,
      margin: '0px 0px',
    }}
  />
)
