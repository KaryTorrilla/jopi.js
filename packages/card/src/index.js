import React from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'

export const Card = ({ variant, active, ...props }) => {
  const variantValues = Object.values(theme.card)[Object.keys(theme.card).indexOf(variant)]
  const backgroundColor = active ? variantValues[':focus'].backgroundColor : variantValues.backgroundColorValue
  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='card'
        variant={variant}
        {...props}
        __css={{
          fontFamily: 'primary',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          backgroundColor: backgroundColor,
          position: 'static',
        }}
        />
    </Box>
  )
}
