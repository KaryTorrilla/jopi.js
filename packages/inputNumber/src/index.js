import React, { useEffect, useRef, useState } from 'react'
import PropTypes from 'prop-types'
import { Box } from '@oneloop/box'
import { Icon } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import theme from '@oneloop/theme'
import '../../theme/styles/globals.css'

export const InputNumber = ({ max = 99, min = 0, startVal = 0, step = 1, val = () => { }, text, variant = 'default', ...props }) => {
  const [valueInput, setValueInput] = useState(startVal)

  useEffect(() => {
    val(valueInput)
  }, [valueInput])

  const handleChange = (e) => {
    if (e.target.value === '') {
      return 0
    }

    if (e.target.value >= min && e.target.value <= max) {
      setValueInput(parseInt(e.target.value))
    } else {
      e.stopPropagation()
    }
  }

  const inputRef = useRef()
  const handleFocus = () => inputRef.current.focus()
  return (
    <Box
      as='div'
      tx="inputNumber"
      variant={variant}
      __css={{
        display: 'inline-flex',
        alignItems: 'center',
        padding: '7px 10px',
        gap: '8px',
        borderRadius: '5px',
      }}
      onClick={handleFocus}
    >
      <Box
        as='input'
        ref={inputRef}
        type='number'
        placeholder='00'
        id='inputNumberDisplay'
        value={parseInt(valueInput).toString()}
        onChange={(e) => handleChange(e)}
        __css={{ width: '22px', background: 'transparent', border: 'none', outline: 'none', color: theme.colors.neutralGray3, textAlign: 'center', padding: '0' }}
        {...props}
        onKeyDown={(e) => {
          if (e.target.value.length === 1 && e.key === 'Backspace') {
            setValueInput(0)
            e.target.value = 0
          }
        }}
      />
      <Text variant='body.fontSize14' style={{ marginRight: '15px' }} >{text}</Text>
      <Box
        __css={{
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        <Icon
          id='arrowUpNumberInput'
          icon='icon-dropdown'
          fontSize='12px'
          style={{ transform: 'rotate(180deg) translateX(1px)', cursor: 'pointer' }}
          onClick={() => setValueInput(Math.min(max, valueInput + step))}
        />
        <Icon
          id='arrowDownNumberInput'
          icon='icon-dropdown'
          fontSize='12px'
          style={{ cursor: 'pointer' }}
          onClick={() => setValueInput(Math.max(min, valueInput - step))}
        />
      </Box>
    </Box >
  )
}

InputNumber.propTypes = {
  max: PropTypes.number,
  min: PropTypes.number,
  startVal: PropTypes.number,
  step: PropTypes.number,
  val: PropTypes.func,
  text: PropTypes.string,
  variant: PropTypes.string,
}
