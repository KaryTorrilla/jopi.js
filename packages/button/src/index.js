import React from 'react'
import { Badge } from '@oneloop/badge'
import { Box } from '@oneloop/box'
import { Checkbox } from '@oneloop/checkbox'
import { Icon } from '@oneloop/icons'
import { Image } from '@oneloop/image'
import { useToggle } from '@oneloop/hooks'
import theme from '@oneloop/theme'
import '@oneloop/fonts'

export const Button = ({ variant, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'normal',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        whiteSpace: 'nowrap',
      }}
    />
  </Box>
)

export const ButtonIcon = ({ icon, variant, text, badgeValue = 0, badgeVariant = 'primary', maxWidth, ...props }) => {
  let fontSize
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return fontSize
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    fontSize = Object.values(theme.buttons)[index].fontSize
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: 'bold',
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          whiteSpace: 'nowrap',
          position: 'relative',
        }}
      >
        { icon && <Icon icon={icon} fontSize={fontSize}/> }
        { badgeValue !== 0 && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{ badgeValue }</Badge> }
        { text && <span>{text}</span> }
      </Box>
    </Box>
  )
}

export const ButtonHoldPress = ({ variant, active = false, text, icon, badgeValue = 0, badgeVariant = 'primary', isCollapsible, hasCheckbox = false, disabled = false, maxWidth, ...props }) => {
  let colorValue
  let colorValueIcon
  let fontSize
  let fontWeight
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index][':focus'] !== undefined && active) {
        colorValue = Object.values(theme.buttons)[index][':focus'].color
        fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
        if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
          colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
        }
      } else if (Object.values(theme.buttons)[index].color !== undefined) {
        colorValue = Object.values(theme.buttons)[index].color
      } else if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
        colorValueIcon = Object.values(theme.buttons)[index].colorIcon
      } else if (Object.values(theme.buttons)[index].fontWeight !== undefined) {
        fontWeight = Object.values(theme.buttons)[index].fontWeight
      }
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return colorValue
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    colorValue = Object.values(theme.buttons)[index].color
    colorValueIcon = Object.values(theme.buttons)[index].colorIcon
    fontSize = Object.values(theme.buttons)[index].fontSize
    fontWeight = Object.values(theme.buttons)[index].fontWeight
    if (active) {
      colorValue = Object.values(theme.buttons)[index][':focus'].color
      colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
      fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
    }
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: {fontWeight},
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          //  justifyContent: 'center',
          width: maxWidth &&'100%',
          alignItems: 'center',
          whiteSpace: 'nowrap',
        }}
      >
        { hasCheckbox && active && <Checkbox defaultChecked disabled={disabled}/>}
        { hasCheckbox && !active && <Checkbox defaultChecked={false} disabled={disabled}/>}
        { icon && <Icon icon={icon} fontSize={fontSize} style={{ color: ( colorValueIcon || colorValue ) }}/> }
        { text && <span style={colorValue && { color: colorValue, fontWeight: fontWeight }}>{text}</span>}
        { badgeValue === 0 && isCollapsible && <span className='icon-dropdown' style={{ position: 'absolute', right: '10px', color: '#798B97', fontSize: '14px', transform: active ? 'rotate(-180deg)' : 'rotate(0deg)', paddingTop: '4px' }}/> }
        { badgeValue !== 0 && !text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{badgeValue}</Badge> }
        { badgeValue !== 0 && text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', right: '16px' }}>{badgeValue}</Badge> }
      </Box>
    </Box>
  )
}

export const ButtonUser = ({ variant, srcImage, text, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'bold',
        fontSize: '14px',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        height: '48px',
        whiteSpace: 'nowrap',
        padding: '13px 26px 13px 26px',
        gap: '8px',
      }}
    >
      {srcImage && <Image src={srcImage} variant="avatar"/> }
      {!srcImage && <span className='icon-contacto' style={{ fontSize: '24px', height: '24px' }}/>}
      {text}
    </Box>
  </Box>
)
