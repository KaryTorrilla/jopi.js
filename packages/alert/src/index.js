import React from 'react'
import { Flex } from '@oneloop/box'
import { Check, Exclamation, Times } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Button } from '@oneloop/button'

export const Alert = ({ type = 'success', text, ...props }) => (
  <Flex
    variant="alert"
    {...props}
    __css={{
      width: '100%',
      bg: 'white',
      borderRadius: 'circle',
      boxShadow: 'active',
      justifyContent: 'space-between',
      p: 1,
    }}
  >
    {type === 'success' && (
      <Check
        bg="success"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'warning' && (
      <Exclamation
        bg="warning"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'danger' && (
      <Times
        bg="danger"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    <Text variant="body.0" display="flex" alignItems="center" flexGrow={1}>
      {text}
    </Text>
    <Button
      size={32}
      bg="#F6F2F2"
      p={3}
      display="inline-flex"
      alignSelf="center"
      mr={2}
    >
      <Times color="#716F6F" m="auto" />
    </Button>
  </Flex>
)
