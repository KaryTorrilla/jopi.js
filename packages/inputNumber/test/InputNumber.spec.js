import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import 'jest-styled-components'

import { InputNumber } from '../src'

Enzyme.configure({ adapter: new Adapter() })

describe('InputNumber', () => {
  it('renders correctly', () => {
    const component = shallow(<InputNumber text="Prueba test" />)

    expect(component).toMatchSnapshot()
  })

  it('add', () => {
    const component = shallow(<InputNumber text="Prueba test" />)

    component.find('#arrowUpNumberInput').simulate('click')
    expect(component.find('#inputNumberDisplay').props().value).toEqual('1')
  })

  it('subtract', () => {
    const component = shallow(<InputNumber text="Prueba test" startVal={10} />)

    component.find('#arrowDownNumberInput').simulate('click')
    expect(component.find('#inputNumberDisplay').props().value).toEqual('9')
  })

  it('max number', () => {
    const component = shallow(<InputNumber text="Prueba test" max={10} />)
    component
      .find('#inputNumberDisplay')
      .simulate('change', { target: { value: '5' } })
      .simulate('change', {
        stopPropagation: () => undefined,
        target: { value: '12' },
      })
    expect(component.find('#inputNumberDisplay').props().value).toEqual('5')
  })

  it('fn val', () => {
    const component = mount(<InputNumber text="Prueba test" val={() => {}} />)
    component
      .find('#inputNumberDisplay')
      .at(1)
      .simulate('change', { target: { value: '2' } })
  })

  it('delete input', () => {
    const component = shallow(<InputNumber text="Prueba test" />)
    component
      .find('#inputNumberDisplay')
      .simulate('change', { target: { value: '' } })
      .simulate('change', { target: { value: '3' } })
      .props()
      .onKeyDown({ key: 'Backspace', target: { value: '1' } })
  })
})
