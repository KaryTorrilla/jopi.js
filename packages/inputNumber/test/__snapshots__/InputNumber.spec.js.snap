// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber renders correctly 1`] = `ShallowWrapper {}`;
